src/Clock/GenericClock.php

```diff
@@ -13,13 +13,28 @@ public function now(): DateTimeImmutable
         return new DateTimeImmutable('now');
     }
 
-    public function time(): int
+    /**
+     * Returns the unix timestamp of the current time in the given unit.
+     *
+     * @param TimeUnit $unit
+     *
+     * @return int
+     */
+    public function time(TimeUnit $unit = TimeUnit::SECOND): int
     {
-        return time();
+        return (int) floor((microtime(true) * 1_000_000) / $unit->toMicroseconds());
```

Comment:
I don't really like this; Maybe we should default to microseconds instead and then add some utility class to convert any unit to microseconds?